sympy/combinatorics/fp_groups.py

```diff
@@ -527,38 +560,25 @@ def subgroup_quotient(G, H, fp_group=None):
     if isinstance(G, list) and not fp_group:
         raise ValueError("The fp_group must be"
                             "defined when the group is a list")
-    free_group = fp_group.G
+    free_group = fp_group.free_group
 
     if not isinstance(fp_group, FpGroup):
         raise ValueError("The parent group must be an instance"
                                     "of FpGroup")
 
-    if not (G.free_group == free_group
-                or H.free_group == free_group):
-        raise ValueError("The elements groups must belong to"
-                            "to the same free group as that of the parent group")
-
-    if isinstance(G, list):
-        # Todo - Set homomorphism to True after the merge
-        g_gens, g_rels, _gens = reidemeister_presentation(fp_group, G, rel=True)
-        _homomorphism = homomorphism(G, fp_group, g_gens, _gens, check=False)
-        g_gens = _gens
-        g_rels = _homomorphism(g_rels)
-    else:
-        g_gens = G.generators
-        g_rels = G.relators
-
-    if isinstance(H, list):
-        # Todo - Set homomorphism to True after the merge
-        h_gens, h_rels, _gens = reidemeister_presentation(fp_group, H, rel=True)
-        _homomorphism = homomorphism(H, fp_group, h_gens, _gens, check=False)
-        h_gens = _gens
-        h_rels = _homomorphism(h_rels)
-    else:
-        h_gens = H.generators
-        h_rels = H.relators
+    if ((isinstance(G, list) and not all(elem in free_group for elem in G))
+        or (isinstance(G, FpGroup) and not (G.free_group == free_group))):
+            raise ValueError("The elements groups must belong to"
+                                    "to the same free group as that of the parent group")
+    if ((isinstance(H, list) and not all(elem in free_group for elem in H))
+        or (isinstance(H, FpGroup) and not (H.free_group == free_group))):
+            raise ValueError("The elements groups must belong to"
+                                    "to the same free group as that of the parent group")
 
-    q_relators = g_gens + h_rels
+    g_gens, g_rels = _get_map(fp_group, G)
+    h_gens, h_rels = _get_map(fp_group, H)
+
+    q_relators = list(g_rels) + list(h_gens)
     q_group = FpGroup(free_group, q_relators)
```

Comment:
This doesn't work when both `G` and `H` are proper subgroups of `fp_group`. For example:
```
>>> F, a, b, c = free_group("a b c")
>>> G = [a,b]
>>> H = [a]
```
Then `G/H` should be isomorphic to a free group on `a` so it should be `<a, b, c | b, c>` but `q_group` will be `<a, b, c | b>` which is a free group on 2 generators.

What you should really do, is find a presentation of `G` along with an injection `T_G` into `fp_group`, translate the generators of `H` into elements of this new presentation (btw, this would also let you check if `H` is a subgroup of `G` which you should do), then return `q_group` on the free group of the presentation of `G`. Does this make sense?

Reply:
>  then return q_group on the free group of the presentation of G

Wouldn't that be a problem if the computed quotients don't belong to the same `FreeGroup` as that of the `FpGroup`?